Fix async and display issue on quick search

paulfalgout opened the pull request:
Shortcut Story ID: [sc-30964] [sc-30961]

Testing the first story in cypress is a bit complicated due to the async request needing to be timed along with the debounce.  At least it was pretty flaky.. maybe a `cy.clock` somehow would help, but nothing at first glance was specific.  That said a real world test was easy and all of the logic branches should be covered here.

nmajor25 reviewed (approved):
Tested in my sandbox, bugs appear to be fixed!

paulfalgout committed:
Abort xhr on cancel

Exposes jquery.xhr.abort on the deferred object.

Also clearing the hasIdentifiers cache, but the crux of the issue was `fetch` on the collection prototype was not cancelable, so preventing the `search` trigger only cancelled some things, but the collection continued to update.

Also clearing the debounce for good measure

paulfalgout committed:
Move header inside scroll area

This inadvertently fixes the esc issue and prevents an odd column alignment issue with the scrollbar

paulfalgout commented:
@nmajor25 last minute revision due to an issue of header alignment with the scrollbar.  mark didn't mind putting the headers in the scroll area to solve the problem.  It could be done with javascript, but a bit overkill for this use case.  This also changed the esc button solution.

<img width="537" alt="Screen Shot 2022-09-22 at 11 57 15 PM" src="https://user-images.githubusercontent.com/2028470/191786188-2ee6b769-4a0e-4f5c-ad0d-2e855d9f8046.png">

paulfalgout commented:
the coveralls issue is a fluke
